### PR TITLE
DDP-5454 limit search query pagination size for auth0 user emails

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Potentially fixes DDP-5454.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

**Comment by yufengwng** [comment]
Data exports for Prion is failing with a stacktrace that looks like this:
```
Error while retrieving auth0 user emails, continuing pagination No content to map due to end-of-input
at [Source: (String)""; line: 1, column: 0]
at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4146)
at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3043)
at com.auth0.net.CustomRequest.createResponseException(CustomRequest.java:103)
at com.auth0.net.CustomRequest.parseResponse(CustomRequest.java:57)
at com.auth0.net.BaseRequest.execute(BaseRequest.java:37)
at org.broadinstitute.ddp.util.Auth0Util.getUserPassConnEmailsByAuth0UserIds(Auth0Util.java:393)
at org.broadinstitute.ddp.export.DataExporter.fetchAndCacheAuth0Emails(DataExporter.java:166)
...
```

This suggests that it's failing during the Auth0 API call to fetch user emails. Specifically, it got an empty response back and the Auth0 library is failing to deserialize it. One can argue that it's likely a bug in the Auth0 library (I haven't looked if there's a new version of library). I did, however, was able to reproduce this locally and I believe we can fix this on our end.

I ran a Prion data export locally pointed at the `test` environment. I confirmed that we are indeed getting back an empty response, which looks something like this (there is nothing after the headers):

Screen Shot 2020-12-17 at 6 13 55 PM

Auth0's [docs](https://auth0.com/docs/api/management/v2#!/Users/get_users) for the API in question specifies a few scenarios for status code 400. We are using the v3 search engine, and we're not running over the record limit since we're always fetching the first page. So by process of elimination, it likely means something is wrong with our query string.

My suspicion is that we're running over the _URL length limit_. I believe browsers (and probably servers too!) have a limit on how long a URL can be. Since this Auth0 API is a `GET` call, the auth0_user_ids that we're searching for are all jammed into the URL query string. Taking a look at the auth0_user_ids in Prion (either test, staging, or production), we see an odd pattern like:
```
auth0|auth0|12345...
```
Typically, auth0_user_ids are `auth0|12345...`. Based on what @charlottedye told me, this was due to an undesirable artifact of how import/export was done for Prion migration. This seems fine for Auth0 -- we're able to lookup and work with the user just fine. However, because there's an extra part of `auth0|` in front of each auth0_user_id of imported Prion users, the query string now becomes longer for requests related to Prion. Coincidentally, we only see this issue with Prion data export and not other studies.

So in short, I believe we're hitting a general URL length limit issue, which was surfaced by the fact that Prion imported users have longer ids. The fix here is to reduce the number of auth0_user_ids we include in each search request.

**Review by ssettipalli — APPROVED** [review]
May be right  way is to clean it up.
Update auth0 userid in database and auth0 to remove extra auth0 from auth0|auth0|5d9247f1a2e6850dfcxxxx . 
Reducing url size should fix the issue though. :+1:

**Comment by yufengwng** [comment]
@ssettipalli I think cleaning up the data will be ideal, but definitely a more extensive operation than what I can cover in a PR. I'll put in a note about it and discuss with Charlotte once she gets back from vacation.
